### PR TITLE
Fix small static analysis items

### PR DESCRIPTION
**Pull request by wolf99**
While testing CppCheck static analysis I see small items that could be fixed in several exercises.
This PR fixes those. Each commit is in relation to items in a different exercise.

### CODE REVIEW

File: exercises/meetup/src/example.c [hunk 1]
```diff
@@ -108,16 +108,15 @@ int meetup_day_of_month(unsigned int year, unsigned int month, const char *week,
...
-   int day_offset;
-   int reference_day = 1;       // used to calculate a baseline day of week given a target date.
 
    int which_week_of_month = get_week_of_month(week);
    int target_day_in_week = get_requested_day_of_week(day_of_week);
 
    // check for valid lookup...
    if ((which_week_of_month >= 0) && (target_day_in_week >= 0)) {
+      int day_offset;
+      int reference_day = 1;    // used to calculate a baseline day of week given a target date.
```
Comment (ryanplusplus) [inline exercises/meetup/src/example.c]: 👍 👍

File: exercises/raindrops/src/example.c [hunk 2]
```diff
@@ -15,7 +15,7 @@ char *convert(char result[], int drops)
       strcat(result, "Plong");
    }
    if (strlen(result) == 0) {
-      char drops_string[12] = "";
+      char drops_string[12] = "\0";
```
Comment (ryanplusplus) [inline exercises/raindrops/src/example.c]: Isn't this equivalent? What was the warning that this fixes?

Comment (wolf99) [reply exercises/raindrops/src/example.c]: The error given is for ln #19 (`strcat(result, drops_string)`). It reads `(error) Buffer is accessed out of bounds`.

Of course the buffer is _not_ accessed out of bounds by the example code. This then means that either the standard library functions used with the buffer directly reference some out of bounds index or this is a false positive. I find the former hard to credit. While searching online on this I came across [this sourcforge bug report](https://sourceforge.net/p/cppcheck/discussion/general/thread/2c33dfc5/) that mentions a similar false positive in the initialization. The fix for that made it into CppCheck 1.81, but the version in the Ubuntu PPA is 1.72. Getting a later version requires building from source so I've not been able to see if a later version fixes this due to time constraints (I have actually had a stab at this but it turned into dependency-fest).

While `= ""` should work, `= "\0";`, `= { '\0' }` or `= { 0 };` are all equally as valid. I'm not sure on the C spec for initialization of empty strings but I do know that for initializer *lists* it mentions something like 

> An initializer-list must have at least one initializer

and also 

> If there are fewer initializers in a brace-enclosed list than there are elements or members of an aggregate, or fewer characters in a string literal used to initialize an array of known size than there are elements in the array, the remainder of the aggregate shall be initialized implicitly the same as objects that have static storage duration

`"\0"` seems to fit the bill with the least amount of change.

Comment (ryanplusplus) [reply exercises/raindrops/src/example.c]: Ah, sounds good. Thanks for the in-depth explanation!

**Comment by ryanplusplus** [comment]
LGTM

**Pull request merged**